kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/ScheduleStrategy.java

```diff
@@ -270,4 +275,22 @@ public void onMessageConfirmed(int messageId, String topic) {
         updateState(State::onMessageEvent);
     }
 
+    @Override
+    public void onPublish() {
+        DataMessage tempDataMessage = this.connectionManager.getNextMessage();
+
+        if (tempDataMessage == null) {
+            return;
+        }
+
+        if (this.isConnectionSchedulePriorityOverrideEnabled
```

Comment:
Maybe we can add a new `State onPublish()` method to the State interface and move this logic there. The only interesting implementation will probably be in the `AwaitConnectTime` state, that will perform a transition to the `AwaitConnect` state if needed